File: app/src/main/java/com/omatt/fdlsandbox/modules/MainContract.java
package com.omatt.fdlsandbox.modules;

import android.content.Intent;

import com.omatt.fdlsandbox.BasePresenter;
import com.omatt.fdlsandbox.BaseView;

/**
 * Created by omarmatthew on 6/27/17.
 * MainActivity View
 */

interface MainContract {
    interface View extends BaseView<Presenter> {
        void updateDynamicLinkLong(String link);

        void updateDynamicLinkShort(String link);
    }
    interface Presenter extends BasePresenter<View> {
        void processDeepLink(Intent intent);

        void buildDynamicLink();
    }
}

File: fdlsandbox-android-kotlin/app/src/main/java/com/omatt/fdlsandbox/modules/MainContract.kt
package com.omatt.fdlsandbox.modules

import android.app.Activity
import android.content.Context
import android.content.Intent
import com.omatt.fdlsandbox.BasePresenter
import com.omatt.fdlsandbox.BaseView

/**
 * Created by omarmatthew on 9/16/2017.
 * Main Contract
 */
interface MainContract {
    interface View : BaseView<Presenter> {
        fun updateDynamicLinkLong(link: String)
        fun updateDynamicLinkShort(link: String)
    }

    interface Presenter : BasePresenter<View> {
        fun sendAppInvite(activity: Activity, resultCode : Int)
        fun processDeepLink(intent: Intent)
        fun buildDynamicLink(context: Context)
    }
}